examples/tracing/src/main.rs

```diff
@@ -0,0 +1,35 @@
+use autometrics::{autometrics, encode_global_metrics, integrations::tracing::AutometricsLayer};
+use tracing::{instrument, trace};
+use tracing_subscriber::{prelude::*, EnvFilter};
+use uuid::Uuid;
+
+// Autometrics looks for a field called `trace_id` and attaches
+// that as an exemplar for the metrics it generates.
+#[autometrics]
+#[instrument(fields(trace_id = %Uuid::new_v4()))]
+fn outer_function() {
+    trace!("Outer function called");
+    inner_function("hello")
+}
+
+// This function will also have the `trace_id` attached as an exemplar
+// because it is called within the same span as `outer_function`.
+#[autometrics]
+#[instrument]
+fn inner_function(param: &str) {
+    trace!("Inner function called");
+}
+
+fn main() {
+    tracing_subscriber::fmt::fmt()
+        .finish()
+        .with(EnvFilter::from_default_env())
+        .with(AutometricsLayer::default())
+        .init();
+
+    for _i in 0..10 {
+        outer_function();
+    }
```

Comment:
The way exemplars work in Prometheus clients (only the last exemplar is ever returned when encoding the metrics for scraping endpoints), technically there's no need to loop `outer_function` calls; unless it's to show non-`1` numbers for the metrics later? We could also probably raise the log levels from `trace!` to `info!` and make sure those logs are visible when running the example, so comparing the trace ID from the log event and the prometheus data is easier.

Reply:
Yeah, the loop was just to show non-1 numbers, but that's not strictly necessary so I can take it out